keras/layers/recurrent_convolutional.py

```diff
@@ -0,0 +1,519 @@
+        input_shape: input_shape
+
+    # Masking
+        This layer supports masking for input data with a variable number
+        of timesteps. To introduce masks to your data,
+        use an [Embedding](embeddings.md) layer with the `mask_zero` parameter
+        set to `True`.
+        **Note:** for the time being, masking is only supported with Theano.
+
+    # TensorFlow warning
+        For the time being, when using the TensorFlow backend,
+        the number of timesteps used must be specified in your model.
+        Make sure to pass an `input_length` int argument to your
+        recurrent layer (if it comes first in your model),
+        or to pass a complete `input_shape` argument to the first layer
+        in your model otherwise.
+
+
+    # Note on using statefulness in RNNs
+        You can set RNN layers to be 'stateful', which means that the states
+        computed for the samples in one batch will be reused as initial states
+        for the samples in the next batch.
+        This assumes a one-to-one mapping between
+        samples in different successive batches.
+
+        To enable statefulness:
+            - specify `stateful=True` in the layer constructor.
+            - specify a fixed batch size for your model, by passing
+                a `batch_input_size=(...)` to the first layer in your model.
+                This is the expected shape of your inputs *including the batch
+                size*.
+                It should be a tuple of integers, e.g. `(32, 10, 100)`.
+
+        To reset the states of your model, call `.reset_states()` on either
+        a specific layer, or on your entire model.
+    '''
+
+    def __init__(self, weights=None,
+                 return_sequences=False, go_backwards=False, stateful=False,
+                 nb_row=None, nb_col=None, nb_filter=None,
+                 dim_ordering=None,
+                 input_dim=None, input_length=None, **kwargs):
+        self.return_sequences = return_sequences
+        self.initial_weights = weights
+        self.go_backwards = go_backwards
+        self.stateful = stateful
+
+        self.nb_row = nb_row
+        self.nb_col = nb_col
+        self.nb_filter = nb_filter
+        self.dim_ordering = dim_ordering
+        self.input_spec = [InputSpec(ndim=5)]
+
+        self.input_dim = input_dim
+        self.input_length = input_length
+        # if self.input_dim:
+        #    kwargs['input_shape'] = (self.input_length, self.input_dim)
+
+        super(RecurrentConv2D, self).__init__(**kwargs)
+
+    def compute_mask(self, input, mask):
+        if self.return_sequences:
+            return mask
+        else:
+            return None
+
+    def get_output_shape_for(self, input_shape):
+
+        if self.dim_ordering == 'th':
+            rows = input_shape[2+1]
+            cols = input_shape[3+1]
+        elif self.dim_ordering == 'tf':
+            rows = input_shape[1+1]
+            cols = input_shape[2+1]
```

Comment:
I copied this part from the Conv2D,
And I just shifted the axis from one to add the time axis before.
I will add one to the number
For dim_ordering, this class is an abstract_class, and dim_ordering
should be given a value in the class that inheritate this class.